### DISCUSSION TITLE
shortcode in page.excerpt

### DISCUSSION THREAD

**Discussion by btrem** [11ty/eleventy#2795] [Q&A]
Is it possible to use a shortcode in a page excerpt? When a shortcode appears at the top of a .md file, it executes in the rendered page and returns code, e.g.:

    <section>
    <h2>my section</h2>
    <div class="myClass">
    <p>foo</p>
    </div>
    </section>

or

    <span data-number="16">foo</span>

But in a post list, where I have a loop that displays only an excerpt of the page (delineated by the quasi comment `<!-- excerpt -->`, the rendered page has the shortcode itself, e.g.,

    <p %="" mypairedshortcode="" %=""></p>
    <p>foo</p>
    <p %="" endmypairedshortcode="" %=""></p>

 
    <p>{% myshortcode
    { myNumber: 16, myText: &quot;foo&quot; }
    %}</p>

Is there a way to get shortcodes working in page excerpts?

**Comment by pdehaan** [comment]
Interesting!
I think the excerpts are provided via the [**gray-matter**](https://github.com/jonschlinkert/gray-matter) module, so I doubt they're being passed through the full 11ty pipeline.
&hellip; but I'm not sure how you could re-render the excerpt if you're expecting excerpts to contain shortcodes. I'm not certain if you could use the [**render**](https://www.11ty.dev/docs/plugins/render/) plugin, or if that might not work if the .md contents have already been converted to HTML.
